dockerd-rootless.sh: if no slirp4netns, try pasta

robmry opened the pull request:
**- What I did**

- related to https://github.com/moby/moby/issues/51143

In the `dockerd-rootless.sh` script, if there's no `slirp4netns` try `pasta` before falling back to VPNkit.

When `pasta` is the network driver, the default port driver is `implicit` instead of `builtin`.

Also (second commit), if `slirp4netns` is selected by env-var, still set its default MTU to 65520.

**- How I did it**

_The notes at the top of the script say "Needs recent version of pasta (2023_12_04)" ... but I'm not sure how to check for that. The reported version number depends on the packager, and I don't know of a command line option to check for (like slirp4netns's check for `--netns-type). As it's no longer that new, perhaps a version check isn't essential ... but can add one if there's a way?_

**- How to verify it**

On a Debian Trixie host, with moby 28.5.1, I tried ...
- with `slirp4netns` installed, it was used as normal
- without it ...
  - `dockerd-rootless.sh` -> ok (running with pasta)
    - started a rootless nginx with `-p 8080:80` and checked it worked
  - `DOCKERD_ROOTLESS_ROOTLESSKIT_PORT_DRIVER=builtin dockerd-rootless.sh` -> error
  - `DOCKERD_ROOTLESS_ROOTLESSKIT_PORT_DRIVER=implicit dockerd-rootless.sh` -> ok
- without slirp4netns or pasta or vpnkit -> error

**- Human readable description for the release notes**
```markdown changelog
- dockerd-rootless.sh: if slirp4netns is not installed, try using pasta (passt).
```

robmry committed:
dockerd-rootless.sh: if no slirp4netns, try pasta

Signed-off-by: Rob Murray <rob.murray@docker.com>

vvoland reviewed (approved):
LGTM, but would love to get a blessing from @AkihiroSuda!

robmry committed:
dockerd-rootless: default MTU 65520 for slirp4netns

When DOCKERD_ROOTLESS_ROOTLESSKIT_MTU is not set, and ...
- DOCKERD_ROOTLESS_ROOTLESSKIT_NET is not set and slirp4netns
  is selected, MTU defaults to 65520.
- DOCKERD_ROOTLESS_ROOTLESSKIT_NET=slirp4netns, MTU defaults
  to 1500.

Change the logic so that, however slirp4netns is selected, MTU
defaults to 65520.

Signed-off-by: Rob Murray <rob.murray@docker.com>

AkihiroSuda reviewed (approved):
Thanks

vvoland reviewed (approved):
LGTM, thanks!
